### PR TITLE
SCANNPM-2 Add support to parse proxy configuration from HTTP[S]_PROXY environment variables

### PR DESCRIPTION
**Pull request by 7PH**
Add support to pass `HTTP[S]_PROXY` as env variable when running the bootstrapper, so that it's used both in NodeJS and in the JVM. 

A known limitation is that the proxy protocol (protocol to use for the proxy, **not the target endpoint**) is lost from the environment variable both for NodeJS and the JVM. We are assuming that the protocol to be used with the proxy is the same as the endpoint, we are not supporting hybrid cases (https proxy with http endpoint, or http proxy with https endpoint).

This PR also changes priorities, as discussed in the slack chanel

### CODE REVIEW

File: src/properties.ts [hunk 1]
```diff
@@ -42,7 +43,6 @@ function getDefaultProperties(): ScannerProperties {
       process.env.HOME ?? process.env.USERPROFILE ?? '',
       SONAR_DIR_DEFAULT,
     ),
-    [ScannerProperty.SonarScannerCliVersion]: SCANNER_CLI_VERSION,
```
Comment (7PH) [inline src/properties.ts:45]: As discussed, dropped this from default props

File: src/properties.ts [hunk 2]
```diff
@@ -289,39 +289,61 @@ function getBootstrapperProperties(startTimestampMs: number): ScannerProperties
 /**
  * Get endpoint properties from scanner properties.
  */
-export function getHostProperties(properties: ScannerProperties): ScannerProperties {
-  let sonarHostUrl = properties[ScannerProperty.SonarHostUrl] ?? '';
-
-  if (!sonarHostUrl || SONARCLOUD_URL_REGEX.exec(sonarHostUrl)) {
+export function getHostProperties(serverUrl?: string, sonarCloudUrl?: string): ScannerProperties {
```
Comment (lucas-paulger-sonarsource) [inline src/properties.ts:292]: what was the rational to move these outside of this method?
```
const sonarHostUrl = properties[ScannerProperty.SonarHostUrl];
const sonarCloudUrl = properties[ScannerProperty.SonarScannerSonarCloudURL];
const hostProperties = getHostProperties(sonarHostUrl, sonarCloudUrl);
```

I might be missing something, but I don't see them used anywhere else

File: src/properties.ts (hunk 1 above)
Comment (lucas-paulger-sonarsource) [reply src/properties.ts:45]: you can drop this from the deps above
https://github.com/SonarSource/sonar-scanner-npm/pull/128/files#diff-c0e4912587622579192cb35813490cac16223f5edb98b8e72d7aed68fc8ffc11L29

File: src/properties.ts [hunk 3]
```diff
@@ -332,30 +354,39 @@ export function getProperties(
...
   }
 
+  // Generate proxy properties from HTTP[S]_PROXY env variables, if not already set
+  const httpProxyProperties = getHttpProxyEnvProperties(
+    userProperties[ScannerProperty.SonarHostUrl],
+  );
+
   // Merge properties respecting order of precedence
-  const properties = {
+  let properties = {
```
Comment (lucas-paulger-sonarsource) [inline src/properties.ts:371]: am I missing something?
```suggestion
  const properties = {
```

**Review by lucas-paulger-sonarsource — APPROVED** [review]
Looks good, some minor cleanup suggestions

**Review by lucas-paulger-sonarsource — CHANGES_REQUESTED** [review]
Oh can you cleanup the config.test.js proxy related tests, and make sure we cover the same scenarios now? 🙇🏻

File: src/properties.ts (hunk 2 above)
Comment (7PH) [reply src/properties.ts:292]: Basically, the idea was to make it clear when reading the code in `getProperties(..)` that this function is "simple" and only expects the server url and SonarCloud URL

```typescript
  const sonarHostUrl = properties[ScannerProperty.SonarHostUrl];
  const sonarCloudUrl = properties[ScannerProperty.SonarScannerSonarCloudURL];
  const hostProperties = getHostProperties(sonarHostUrl, sonarCloudUrl);
// vs
  const hostProperties = getHostProperties(properties);
```

Taking a second look at this and given your doubts, this choice is probably questionable in that it also adds a bit of complexity to the code

File: src/properties.ts (hunk 3 above)
Comment (7PH) [reply src/properties.ts:371]: nope, thanks fixed it 😁

**Review by vdiez — APPROVED** [review]
LGTM

**Pull request merged**